src/lib/cache/TTLCache.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+export interface TTLCacheItem<T> {
+  value: T;
+  expire: number;
+}
+
+export type TTLCacheOptions = {
+  ttl: number,
+};
+
+class TTLCache<K, V> {
+  cache: Map<K, TTLCacheItem<V>> = new Map<K, TTLCacheItem<V>>();
+  options: TTLCacheOptions;
+
+  constructor(options?: TTLCacheOptions) {
+    this.options = {
+      ttl: Infinity,
+      ...options,
+    };
+  }
+
+  set(key: K, value: V, ttl: number = this.options.ttl, now: number = Date.now()): boolean {
+    // Adding the value to the cache is not possible if ttl is zero.
+    if (ttl === 0) {
+      return false;
+    }
+
+    // Check for infinity in which case the item persists forever.
+    const expire = ttl < Infinity ? now + ttl : Infinity;
+
+    const item: TTLCacheItem<V> = {
+      value,
+      expire,
+    };
+
+    this.cache.set(key, item);
+
+    return true;
+  }
+
+  get(key: K, remove: boolean = false, now?: number): ?V {
+    const value = this.touch(key, now);
+
+    if (value == null) {
+      return null;
+    }
+
+    if (remove) {
+      this.remove(key);
+    }
+
+    return value;
+  }
+
+  has(key: K): boolean {
+    return this.cache.has(key);
+  }
+
+  remove(key: K): boolean {
+    return this.cache.delete(key);
+  }
+
+  touch(key: K, now: number = Date.now()): ?V {
+    const item = this.cache.get(key);
+
+    if (item == null) {
+      return null;
+    }
+
+    if (item.expire < now) {
+      this.remove(key);
+
+      return null;
+    }
+
+    return item.value;
+  }
+
+  touchAll(now?: number): void {
+    for (const key of this.cache.keys()) {
```

Comment:
How about this?

```suggestion
this.cache.keys().forEach((key) =>  this.touch(key, now));
```

…as this does not use the regenerator runtime

Reply:
I thought the regenerator runtime is only included when used with Generator or async functions? The generator class itself is not needing the polyfill AFAIK. Will investigate.

Also the regenerator runtime is already included because of the async functions when fetching intial props.

Reply:
https://github.com/airbnb/javascript/issues/1271 has a long discussion of many aspects of this if you're interested :)

Reply:
(Also, you seem to be right, and to have that for loop there is totally okay.)